silverback/runner.py

```diff
@@ -63,15 +67,79 @@ async def run(self):
         await self.app.broker.shutdown()
 
 
-class LiveRunner(BaseRunner):
+class WebsocketRunner(BaseRunner, ManagerAccessMixin):
     """
-    Run a single app against a live network using a basic in-memory queue.
+    Run a single app against a live network using a basic in-memory queue and websockets.
     """
 
     def __init__(self, app: SilverBackApp, *args, **kwargs):
         super().__init__(app, *args, **kwargs)
         logger.info(f"Using {self.__class__.__name__}: max_exceptions={self.max_exceptions}")
 
+        # Check for websocket support
+        provider = app.chain_manager.provider
+        if isinstance(provider, UpstreamProvider):
+            uri = provider.connection_str
+        elif hasattr(provider, "uri"):
+            # Most Web3Providers
+            uri = provider.uri
+        else:
+            raise
+
+        self.ws_uri = uri.replace("http", "ws")
```

Comment:
didnt we add a property to the provider for this?

Reply:
Yes! This PR was from a month ago, so i will refactor for that now